compiler/optimizer/Inliner.cpp

```diff
@@ -2354,45 +2354,58 @@ TR_ParameterToArgumentMapper::initialize(TR_CallStack *callStack)
             else
                {
                TR::SymbolReference * symRef = NULL;
-
+               const static bool disableUseKnownObjectTempsForParms = feGetEnv("TR_DisableUseKnownObjectTempsForParmsInCallee") ? true: false;
+               int argOrdinal = argIndex - _callNode->getFirstArgumentIndex();
+               TR_PrexArgument * prexArgument = _argInfo->get(argOrdinal);
+               // use known object temp if the argument is a known object
+               if (!disableUseKnownObjectTempsForParms && prexArgument
+                   && !parmMap->_parmIsModified
+                   && TR_PrexArgument::knowledgeLevel(prexArgument) == KNOWN_OBJECT
+                   && !prexArgument->isTypeInfoForInlinedBody())
```

Comment:
What is this ```isTypeInfoForInlinedBody``` check for ? Might be good to capture in the comment so that the reasoning is apparent in the code.